src/app/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,237 @@
+  Tab,
+  Tabs,
+  TabTitleText,
+  TextInput,
+} from '@patternfly/react-core';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { TimezonePicker } from './TimezonePicker';
+
+export interface DateTimePickerProps {
+  onSelect?: (date: Date, timezone: Timezone) => void;
+  prefilledDate?: Date; // timezone is ignored
+  onDismiss: () => void;
+}
+
+export const DateTimePicker: React.FC<DateTimePickerProps> = ({ onSelect, onDismiss, prefilledDate }) => {
+  const [t] = useTranslation();
+  const [dayjs, _] = useDayjs();
+  const [activeTab, setActiveTab] = React.useState('date');
+  const [datetime, setDatetime] = React.useState<Date>(new Date());
+  const [timezone, setTimezone] = React.useState<Timezone>(defaultDatetimeFormat.timeZone); // Not affected by user preferences
+
+  const handleTabSelect = React.useCallback((_, key: string | number) => setActiveTab(`${key}`), [setActiveTab]);
+
+  const handleSubmit = React.useCallback(() => {
+    onSelect && onSelect(datetime, timezone);
+  }, [datetime, timezone, onSelect]);
+
+  const handleCalendarSelect = React.useCallback(
+    (date: Date) => {
+      setDatetime((old) => {
+        const wrappedOld = dayjs(old);
+        return dayjs(date).hour(wrappedOld.hour()).minute(wrappedOld.minute()).second(wrappedOld.second()).toDate();
+      });
+      setActiveTab('time'); // Switch to time
+    },
+    [setDatetime, setActiveTab, dayjs]
+  );
+
+  const handleHourChange = React.useCallback(
+    (hrIn24h: number) => {
+      setDatetime((old) => dayjs(old).hour(hrIn24h).toDate());
+    },
+    [setDatetime, dayjs]
+  );
+
+  const handleMinuteChange = React.useCallback(
+    (m: number) => {
+      setDatetime((old) => dayjs(old).minute(m).toDate());
+    },
+    [setDatetime, dayjs]
+  );
+
+  const handleSecondChange = React.useCallback(
+    (s: number) => {
+      setDatetime((old) => dayjs(old).second(s).toDate());
+    },
+    [setDatetime, dayjs]
+  );
+
+  const handleMeridiemChange = React.useCallback(
+    (isAM: boolean) => {
+      setDatetime((old) => {
+        const oldDayjs = dayjs(old);
+        if (isAM !== isHourIn24hAM(oldDayjs.hour())) {
+          return (isAM ? oldDayjs.subtract(12, 'hour') : oldDayjs.add(12, 'hour')).toDate();
+        }
+        return old;
+      });
+    },
+    [setDatetime, dayjs]
+  );
+
+  const selectedDatetimeDisplay = React.useMemo(() => dayjs(datetime).format('L LTS'), [datetime, dayjs]);
+
+  React.useEffect(() => {
+    if (prefilledDate) {
+      setDatetime(new Date(prefilledDate));
+    }
+  }, [setDatetime, prefilledDate]);
+
+  return (
+    <Form>
+      <Tabs
+        aria-label={t('DateTimePicker.ARIA_LABELS.TABS') || ''}
+        onSelect={handleTabSelect}
+        activeKey={activeTab}
+        isFilled
+        role="region"
+      >
+        <Tab key={'date'} eventKey={'date'} title={<TabTitleText>{t('DATE', { ns: 'common' })}</TabTitleText>}>
+          <FormGroup key={'calendar'}>
```

Comment:
Should the `'calendar'` here and the `'time'` below be extracted to constants, or a typedef for them declared?